### PR TITLE
Fix issue causing asset dependencies to be ignored

### PR DESCRIPTION
**Pull request by pbbastian**
# Purpose of this PR

Import dependencies from the `Generator` class were being ignored due to re-assignment of list without `out`. I changed it to `AddRange` on the input list. Also bumped the importer version so that all Shader Graphs get re-imported and then have the dependencies setup correctly.

The issue was introduced in this PR: https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/4940

# Testing status
## Manual Tests
- Reproduced issue locally
- Verified fix locally

To reproduce the issue, you can make any change to `com.unity.render-pipelines.universal/Editor/ShaderGraph/Targets/UniversalTarget.cs` and notice that without the fix it does not cause a re-import of Shader Graphs, even though it registers itself as a source dependency.

You can also modify put a `Debug.Log` in here to see what ends up getting declared, and notice that without the fix it does not declare anything.
https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.shadergraph/Editor/Importers/ShaderGraphImporter.cs#L168

### CODE REVIEW

File: com.unity.shadergraph/Editor/Importers/ShaderGraphImporter.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ namespace UnityEditor.ShaderGraph
     // This ifdef can be removed once V2 is the only option.
     [ScriptedImporter(101, Extension, 3)]
```
Comment (marctem) [inline com.unity.shadergraph/Editor/Importers/ShaderGraphImporter.cs:23]: Gotta bump both versions; Anything using Hybrid will have the top version number and those will need to re-import as well.

Comment (marctem) [inline com.unity.shadergraph/Editor/Importers/ShaderGraphImporter.cs:23]: Gotta bump both versions; Anything using Hybrid will have the top version number and those will need to re-import as well.

**Review by marctem — CHANGES_REQUESTED** [review]

File: com.unity.shadergraph/Editor/Importers/ShaderGraphImporter.cs (hunk 1 above)
Comment (pbbastian) [reply com.unity.shadergraph/Editor/Importers/ShaderGraphImporter.cs]: ahh yeah, good point!

**Review by marctem — APPROVED** [review]

**Comment by pbbastian** [comment]
@Unity-Technologies/gfx-qa-bellevue Any update on this?

**Comment by LandonTownsendUnity** [comment]
I can not get the debug.log to print anything. I put it where you said, and then made a change to UniversalTarget.cs on your branch, and it did not cause the debug.log to happen. Am I missing something?

**Review by cdxntchou — APPROVED** [review]

**Review by ghost — APPROVED** [review]
Changed the location I put the debug log. It doesn't reimport on master, but I've confirmed it does on this branch, when UniversalTarget.cs is changed. I'll approve.

**Pull request merged**